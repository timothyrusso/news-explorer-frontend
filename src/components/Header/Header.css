.header {
  background-image: url(../../images/header_background.png);
  width: 100%;
  min-height: 576px;
}

.header__title {
  margin: 0 auto;
  padding: 160px 0 32px;
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: 400;
  font-size: 60px;
  line-height: 1.07;
  color: #fff;
  max-width: 608px;
}

.header__subtitle {
  margin: 0 auto;
  padding-bottom: 88px;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  color: #fff;
  max-width: 608px;
}

@media screen and (max-width: 768px) {
  .header {
    min-height: 400px;
  }

  .header__title {
    font-size: 36px;
    line-height: 1.11;
    max-width: 385px;
    margin-left: 160px;
    padding: 104px 0 16px;
  }

  .header__subtitle {
    font-size: 18px;
    line-height: 1.33;
    max-width: 452px;
    margin-left: 160px;
    padding-bottom: 48px;
  }
}
