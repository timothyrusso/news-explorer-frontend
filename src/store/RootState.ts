import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
