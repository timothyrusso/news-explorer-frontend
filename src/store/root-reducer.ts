import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
