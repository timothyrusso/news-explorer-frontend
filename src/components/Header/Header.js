import "./Header.css";
import SearchForm from "../SearchForm/SearchForm";

const Header = ({ startLoadingNews, activateSearch }) => {
  return (
    <header className="header">
      <h1 className="header__title">What's going on in the world?</h1>
      <p className="header__subtitle">
        Find the latest news on any topic and save them in your personal
        account.
      </p>
      <SearchForm
        startLoadingNews={startLoadingNews}
        activateSearch={activateSearch}
      />
    </header>
  );
};

export default Header;
