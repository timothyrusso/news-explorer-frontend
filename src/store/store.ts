import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
