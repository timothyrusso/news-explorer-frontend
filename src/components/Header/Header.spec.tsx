import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Header from './Header';

describe('Header component', () => {
  it('Should render the Header component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Header />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
