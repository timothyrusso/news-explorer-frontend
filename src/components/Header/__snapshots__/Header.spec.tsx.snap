// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render the Header component 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h1
      class="header__title"
    >
      What's going on in the world?
    </h1>
    <p
      class="header__subtitle"
    >
      Find the latest news on any topic and save them in your personal account.
    </p>
    <form
      class="search-form"
    >
      <input
        class="search-form__input "
        id="search-input"
        name="search"
        placeholder="Enter topic"
        type="text"
        value=""
      />
      <button
        class="search-form__button"
      >
        Search
      </button>
    </form>
  </header>
</DocumentFragment>
`;
